packages/images-plugin/lib/finder.js

```diff
@@ -103,34 +107,30 @@ export default class ImageElementFinder {
    */
   registerImageElement(imgEl) {
     this.imgElCache.set(imgEl.id, imgEl);
-    const protoKey = this.driver.isW3CProtocol() ? W3C_ELEMENT_KEY : MJSONWP_ELEMENT_KEY;
-    return imgEl.asElement(protoKey);
+    if (!this.driver) {
+      throw new ReferenceError('No driver set!');
+    }
+    return imgEl.asElement();
   }
 
-  /**
-   * @typedef FindByImageOptions
-   * @property {boolean} [shouldCheckStaleness=false] - whether this call to find an
-   * image is merely to check staleness. If so we can bypass a lot of logic
-   * @property {boolean} [multiple=false] - Whether we are finding one element or
-   * multiple
-   * @property {boolean} [ignoreDefaultImageTemplateScale=false] - Whether we
-   * ignore defaultImageTemplateScale. It can be used when you would like to
-   * scale b64Template with defaultImageTemplateScale setting.
-   */
-
   /**
    * Find a screen rect represented by an ImageElement corresponding to an image
    * template sent in by the client
    *
+   * @template {boolean} [Multiple=false]
+   * @template {boolean} [CheckStaleness=false]
    * @param {string} b64Template - base64-encoded image used as a template to be
    * matched in the screenshot
-   * @param {FindByImageOptions} opts - additional options
-   *
-   * @returns {Promise<Element|Element[]|ImageElement>} - WebDriver element with a special id prefix
+   * @param {FindByImageOptions<Multiple,CheckStaleness>} opts - additional options
+   * @returns {Promise<FindByImageResult<Multiple,CheckStaleness>>} - WebDriver element with a special id prefix
    */
   async findByImage(
     b64Template,
-    {shouldCheckStaleness = false, multiple = false, ignoreDefaultImageTemplateScale = false}
+    {
+      shouldCheckStaleness = /** @type {CheckStaleness} */ (false),
```

Comment:
these are simple booleans

Reply:
yes, but we can't use conditional types without them (the things that look like ternary statements).  the type of the return value depends on whether or not these are true or false, and there's no other way to reference the type. [this explains it better than I can](https://www.typescriptlang.org/docs/handbook/2/conditional-types.html)